src/shared/Table/TableView.tsx

```diff
@@ -198,37 +208,52 @@ export const TableView = <TRow, TCol>({
               <OverflowMenuContent isPersistent>
                 <OverflowMenuGroup isPersistent groupType="button">
                   <OverflowMenuItem>
-                    {actions.map((a, idx) => (
-                      <Button
-                        key={idx}
-                        variant={a.isPrimary ? "primary" : undefined}
-                        onClick={a.onClick}
-                      >
-                        {a.label}
-                      </Button>
-                    ))}
+                    {actions.map(
+                      (a, idx) =>
+                        !a.hasKebabToolbarAction && (
+                          <Button
+                            key={idx}
+                            variant={a.isPrimary ? "primary" : undefined}
+                            onClick={a.onClick}
+                          >
+                            {a.label}
+                          </Button>
+                        )
+                    )}
                   </OverflowMenuItem>
                 </OverflowMenuGroup>
               </OverflowMenuContent>
-              <OverflowMenuControl>
-                <Dropdown
-                  isPlain
-                  toggle={<KebabToggle onToggle={toggleIsActionsOpen} />}
-                  isOpen={isActionsOpen}
-                  dropdownItems={actions.map((a, idx) => (
-                    <OverflowMenuDropdownItem
-                      key={idx}
-                      onClick={() => {
-                        a.onClick();
-                        toggleIsActionsOpen(false);
-                      }}
-                    >
-                      {a.label}
-                    </OverflowMenuDropdownItem>
-                  ))}
-                  isFlipEnabled
-                  menuAppendTo="parent"
-                />
+              <OverflowMenuControl
+                hasAdditionalOptions={isToolbarKebabActionVisible}
```

Comment:
I guess to can use actions.some((a) => a.dropdownItems) to check hasAdditionalOptions props instead of passing a prop